### PR TITLE
I got error: Expected argument of type \"null\", \"Symfony\\Component…

### PR DESCRIPTION
**Pull request by yapro**
I got error: 'Expected argument of type "null", "Symfony\Component\HttpFoundation\Request" given'

https://github.com/symfony/form/blob/master/NativeRequestHandler.php#L55

```
"class": "'Symfony\\Component\\Form\\Exception\\UnexpectedTypeException'",
"file": "'\/home\/www\/vendor\/symfony\/form\/NativeRequestHandler.php'",
"line": "56",
"level": 500,
"level_name": "CRITICAL",
```
#0 /vendor/symfony/form/Form.php(499): Symfony\Component\Form\NativeRequestHandler->handleRequest(Object(Symfony\Component\Form\Form), Object(Symfony\Component\HttpFoundation\Request))
#1 /application/controllers/product.php(335): Symfony\Component\Form\Form->handleRequest(Object(Symfony\Component\HttpFoundation\Request))

### CODE REVIEW

File: components/form/introduction.rst [hunk 1]
```diff
@@ -556,7 +556,7 @@ method:
 
         $request = Request::createFromGlobals();
 
-        $form->handleRequest($request);
+        $form->handleRequest();
```
Comment (ogizanagi) [inline components/form/introduction.rst]: As a `Request` instance is created in this example, I guess the fix should be to add the `HttpFoundation` integration before those lines and revert this change:
```php
use Symfony\Component\Form\Forms;
use Symfony\Component\Form\Extension\HttpFoundation\HttpFoundationExtension;

$formFactory = Forms::createFormFactoryBuilder()
    ->addExtension(new HttpFoundationExtension())
    ->getFormFactory();
```

This will add the `HttpFoundationRequestHandler`, instead of using the base `NativeRequestHandler`.

File: components/form/introduction.rst [hunk 2]
```diff
@@ -582,7 +582,7 @@ method:
                 ->add('dueDate', 'date')
                 ->getForm();
 
-            $form->handleRequest($request);
+            $form->handleRequest();
```
Comment (ogizanagi) [inline components/form/introduction.rst]: This should be reverted IMO, as this example shows the usage of the form component inside the Symfony framework. The `HttpFoundationRequestHandler` is registered by default.

File: components/form/introduction.rst [hunk 3]
```diff
@@ -102,7 +102,7 @@ object to read data off of the correct PHP superglobals (i.e. ``$_POST`` or
     Now, when you process a form, you can pass the :class:`Symfony\\Component\\HttpFoundation\\Request`
     object to :method:`Symfony\\Component\\Form\\Form::handleRequest`::
 
-        $form->handleRequest($request);
+        $form->handleRequest();
```
Comment (ogizanagi) [inline components/form/introduction.rst]: This shows the integration with the `HttpFoundation` component. It registers a `HttpFoundationRequestHandler` which needs a `Request` instance.

**Comment by ogizanagi** [comment]
@yapro : I think there is a misunderstanding. 
Do you use the Symfony Framework, or do you use the form component standalone ?
Did you [configure the `HttpFoundation` integration](http://symfony.com/doc/current/components/form/introduction.html#request-handling) properly if you're using the [`HttpFoundation`](http://symfony.com/doc/current/components/http_foundation/introduction.html) component with the standalone form component ?

**Comment by yapro** [comment]
Thx, I use the form component as standalone.
